packages/@glimmer/runtime/lib/scope.ts

```diff
@@ -43,21 +43,13 @@ export function isScopeReference(s: ScopeSlot): s is Reference {
 
 export class PartialScopeImpl implements PartialScope {
   static root(self: Reference<unknown>, size = 0, owner: Owner): PartialScope {
-    let refs: Reference<unknown>[] = new Array(size + 1);
-
-    for (let i = 0; i <= size; i++) {
-      refs[i] = UNDEFINED_REFERENCE;
-    }
+    let refs: Reference<unknown>[] = new Array(size + 1).fill(UNDEFINED_REFERENCE, 0, size);
```

Comment:
Same, I am inclined to think this is not a problem unless there are new sources that indicates otherwise

Reply:
@chancancode it's 70% faster https://jsben.ch/bDWmT

Reply:
Ok sure, yeah it's fine in this case when we are filling it with the same value (though I don't think we should be using that site for benchmarks, and that we don't trust microbenchmarks in general).

What I was trying to say is that you don't have to do this _in general_, when you are going to initialize the array with non-uniform data anyway:

```js
const array = new Array(size + 1);

for (let i = 0; i<size+1; i++) {
  // this should be just fine
  array[i] = Math.random();
}
```

Reply:
@chancancode agree! but in exact case data will be filled with only one value: `UNDEFINED_REFERENCE`

Reply:
The new code seems buggy though. The previous code was iterating up to `<= size` (as opposed to the normal `< size`) to account for the `+ 1`. So I think you just want to fill the whole array normally as opposed to deliberately under-fill it (which seemed very suspect).

Reply:
@chancancode agree! Fixed in both cases